src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationKeyVaultOptions.cs

```diff
@@ -34,5 +37,23 @@ public AzureAppConfigurationKeyVaultOptions Register(SecretClient secretClient)
             SecretClients.Add(secretClient);
             return this;
         }
+
+        /// <summary>
+        /// Sets the callback used to resolve key vault references that have no registered <see cref="SecretClient"/>.
+        /// </summary>
+        /// <param name="secretResolver">A callback which returns a value for the given Key Vault reference.</param>
+        /// <remarks>
+        /// The input to the callback will be a <see cref="Uri"/> in key vault reference format: "https://{vault-name}.vault.azure.net/{type}/{name}/{version}". 
```

Comment:
>{type} [](start = 135, length = 6)

The type would always be "secrets" since this method is specific to secrets.

Reply:
Not necessarily - the `GetSecretAsync` API can resolve certificates too. And we allow users to create references to certificates in App Config.
For example, both of these highlighted references resolve to the same certificate:

![image](https://user-images.githubusercontent.com/5563494/98989992-bbfa1300-24de-11eb-858b-bba59e59b93a.png)

Reply:
`GetSecretAsync` API only resolves secrets. Each certificate in key vault consists of certificate metadata, key and secret, and only the secret part is retrieved. The certificate URL works since we don't have validation on the URI. Even if you change "/certificates/" to "/dummy/", the configuration provider would still fetch the secret.